Sources/Sentry/SentryHttpTransport.m

```diff
@@ -184,35 +191,67 @@ - (void)sendCachedEventsAndEnvelopes {
+            initStoreRequestWithDsn:self.options.dsn
+            andData:fileContents.contents
+            didFailWithError:nil];
+            [self sendCached:request withFilePath:fileContents.path];
+        }
+    }
+}
+
+- (void)sendAllCachedEnvelopes {
+    for (SentryFileContents *fileContents in [self.fileManager getAllEnvelopes]) {
+        SentryEnvelope *envelope = [SentrySerialization envelopeWithData:fileContents.contents];
+        if (nil == envelope) {
+            [self.fileManager removeFileAtPath:fileContents.path];
+            continue;
+        }
         
+        SentryEnvelope *rateLimitedEnvelope = [self.envelopeRateLimit removeRateLimitedItems:envelope];
+        if (rateLimitedEnvelope.items.count == 0) {
+            [self.fileManager removeFileAtPath:fileContents.path];
+            continue;
+        }
         
-        // TODO: Check RateLimit for EnvelopeItemType
+        NSError *requestError = nil;
+        NSURLRequest *request = [self createEnvelopeRequest:envelope didFailWithError:requestError];
         
-        // TODO: Get EventType from event and not use SentryEnvelopeItemTypeEvent
-        NSString *category = [SentryRateLimitCategoryMapper mapEventTypeToCategory:SentryEnvelopeItemTypeEvent];
-        if (![self isReadyToSend:category]) {
+        if (nil != requestError) {
+            [SentryLog logWithMessage:requestError.localizedDescription andLevel:kSentryLogLevelError];
             [self.fileManager removeFileAtPath:fileContents.path];
+            continue;
         } else {
-            SentryNSURLRequest *request = [[SentryNSURLRequest alloc] initStoreRequestWithDsn:self.options.dsn
-                                                                                      andData:fileContents.contents
-                                                                             didFailWithError:nil];
-            
-            
-            [self sendRequest:request withCompletionHandler:^(NSHTTPURLResponse *_Nullable response, NSError *_Nullable error) {
-                // TODO: How does beforeSend work here
-                // We want to delete the event here no matter what (if we had an internet connection)
-                // since it has been tried already.
-                if (response != nil) {
-                    [self.fileManager removeFileAtPath:fileContents.path];
-                }
-
-                dispatch_group_leave(dispatchGroup);
-            }];
+            [self sendCached:request withFilePath:fileContents.path];
         }
     }
 }
 
+- (void)sendCached:(NSURLRequest *)request withFilePath:(NSString *)filePath {
+    [self sendRequest:request withCompletionHandler:^(NSHTTPURLResponse *_Nullable response, NSError *_Nullable error) {
+        // TODO: How does beforeSend work here
+        
+        // If the response is not nil we had an internet connection and we can
+        // delete the cached event or envelope no an error happened or not.
```

Comment:
m: internet connection should not dispose envelopes, only if there's an http error.
no connection should cache or keep in the disk to be sent later on

Reply:
Very good point. This logic was already here. Keep in mind this is only for cached envelopes. If we want to change it, I would do it in another PR. What do you think @HazAT?

Reply:
Sorry am I missing someing, we check `if (response != nil) {` below and only delete it if it isn't nil.

Reply:
yep I focused on the comments because it does something different looks like